modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm

```diff
@@ -713,6 +713,28 @@ sub dump_phenotypes {
   }
 }
 
+sub clean_phenotype_tables {
+  my ($self) = @_;
+
+  my $delete_pheno_feature = qq{
+    DELETE from phenotype WHERE phenotype_id NOT IN (SELECT phenotype_id from phenotype_feature);
+  };
+   
+  my $delete_pheno_ontology_accesion = qq {
+    DELETE from phenotype_ontology_accession WHERE phenotype_id NOT IN (SELECT phenotype_id from phenotype_feature);
```

Comment:
As this query is only run after unused phenotypes have been deleted, you could select them from the phenotype table:
`DELETE from phenotype_ontology_accession WHERE phenotype_id NOT IN (SELECT phenotype_id from phenotype);`